support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/StageStepExecution.java

```diff
@@ -222,18 +222,26 @@ private static void cleanUp(Job<?,?> job, String jobName) {
     }
 
     private static void println(StepContext context, String message) {
+        if (!context.isReady()) {
```

Comment:
Really it would be preferable to send all calls to `get`, `onSuccess`, and `onFailure` to a per-context asynchronous queue, so that the `StageStepExecution.class` monitor is never held while these calls are made. Seems overly complex for now.